clang/test/SemaSYCL/array-kernel-param.cpp

```diff
@@ -55,40 +55,47 @@ int main() {
 // CHECK-NEXT: MemberExpr {{.*}}__init
 
 // Check kernel_B parameters
-// CHECK: FunctionDecl {{.*}}kernel_B{{.*}} 'void (wrapped_array)'
-// CHECK-NEXT: ParmVarDecl {{.*}} 'wrapped_array'
+// CHECK: FunctionDecl {{.*}}kernel_B{{.*}} 'void (int, int)'
+// CHECK-NEXT: ParmVarDecl {{.*}} used _arg_ 'int'
+// CHECK-NEXT: ParmVarDecl {{.*}} used _arg_ 'int'
+// Check kernel_B inits
 // CHECK-NEXT: CompoundStmt
 // CHECK-NEXT: DeclStmt
-// CHECK-NEXT: VarDecl
+// CHECK-NEXT: VarDecl {{.*}} cinit
 // CHECK-NEXT: InitListExpr
-// CHECK-NEXT: ArrayInitLoopExpr {{.*}} 'int [100]'
+// CHECK-NEXT: InitListExpr {{.*}} 'int [2]'
+// CHECK: ImplicitCastExpr
+// CHECK: DeclRefExpr {{.*}} 'int' lvalue ParmVar {{.*}} '_arg_' 'int'
+// CHECK: ImplicitCastExpr
+// CHECK: DeclRefExpr {{.*}} 'int' lvalue ParmVar {{.*}} '_arg_' 'int'
 
-// Check kernel_C parameters
-// CHECK: FunctionDecl {{.*}}kernel_C{{.*}} 'void (struct {{.*}}, __global int *, cl::sycl::range<1>, cl::sycl::range<1>, cl::sycl::id<1>, __global int *, cl::sycl::range<1>, cl::sycl::range<1>, cl::sycl::id<1>, __global int *, cl::sycl::range<1>, cl::sycl::range<1>, cl::sycl::id<1>, __global int *, cl::sycl::range<1>, cl::sycl::range<1>, cl::sycl::id<1>)'
-// CHECK-NEXT: ParmVarDecl {{.*}} 'struct {{.*}}'
-// CHECK-NEXT: ParmVarDecl {{.*}} used _arg_member_acc '__global int *'
-// CHECK-NEXT: ParmVarDecl {{.*}} used _arg_member_acc 'cl::sycl::range<1>'
-// CHECK-NEXT: ParmVarDecl {{.*}} used _arg_member_acc 'cl::sycl::range<1>'
-// CHECK-NEXT: ParmVarDecl {{.*}} used _arg_member_acc 'cl::sycl::id<1>'
-// CHECK-NEXT: ParmVarDecl {{.*}} used _arg_member_acc '__global int *'
-// CHECK-NEXT: ParmVarDecl {{.*}} used _arg_member_acc 'cl::sycl::range<1>'
-// CHECK-NEXT: ParmVarDecl {{.*}} used _arg_member_acc 'cl::sycl::range<1>'
-// CHECK-NEXT: ParmVarDecl {{.*}} used _arg_member_acc 'cl::sycl::id<1>'
-// CHECK-NEXT: ParmVarDecl {{.*}} used _arg_member_acc '__global int *'
-// CHECK-NEXT: ParmVarDecl {{.*}} used _arg_member_acc 'cl::sycl::range<1>'
-// CHECK-NEXT: ParmVarDecl {{.*}} used _arg_member_acc 'cl::sycl::range<1>'
-// CHECK-NEXT: ParmVarDecl {{.*}} used _arg_member_acc 'cl::sycl::id<1>'
-// CHECK-NEXT: ParmVarDecl {{.*}} used _arg_member_acc '__global int *'
-// CHECK-NEXT: ParmVarDecl {{.*}} used _arg_member_acc 'cl::sycl::range<1>'
-// CHECK-NEXT: ParmVarDecl {{.*}} used _arg_member_acc 'cl::sycl::range<1>'
-// CHECK-NEXT: ParmVarDecl {{.*}} used _arg_member_acc 'cl::sycl::id<1>'
+// Correct and enable after struct members are extracted into separate parameters
+// C HECK kernel_C parameters
```

Comment:
suggest to file a github issue and reference it from here

Reply:
PR#1877 is already in progress to address this. 
Existing related GitHub issues - https://github.com/intel/llvm/issues/488, https://github.com/intel/llvm/issues/28